pkg/metrics/proxy/injector/deployment_patch.go

```diff
@@ -59,6 +60,23 @@ func (i *injector) getDeploymentPatchOperations(
 			Value: "true",
 		})
 
+		// Add or update "osiris.deislabs.io/ignoredPaths"
+		if ignoredPaths :=
+			kubernetes.GetIgnoredPaths(deployment.Annotations); len(ignoredPaths) > 0 { // nolint: lll
+			var op string
+			if _, ok :=
+				deployment.Spec.Template.Annotations[kubernetes.IgnoredPathsAnnotationName]; ok { // nolint: lll
+				op = "replace"
+			} else {
+				op = "add"
+			}
+			patchOps = append(patchOps, kubernetes.PatchOperation{
+				Op:    op,
+				Path:  "/spec/template/metadata/annotations/osiris.deislabs.io~1ignoredPaths", // nolint: lll
+				Value: strings.Join(ignoredPaths, ","),
+			})
+		}
+
```

Comment:
I'm having a hard time understanding what this bit of patching accomplishes. I might be missing something, but it looks as if it's patching the deployment's ignored paths annotation with values retrieved _from_ the deployment's ignored paths annotation. Isn't this like saying let x = x?

Reply:
Nevermind. I see it now. You're copying the annotation from the deployment to the the pod spec template within the deployment.

Reply:
in fact it's reading from the deployment's annotations, and writing to the pod's annotations. because later to inject the proxy, we will only have access to the pod's annotations, not the deployment's annotations